spec/factories/vouchers.rb

```diff
@@ -0,0 +1,28 @@
+FactoryBot.define do
+  factory :voucher do
+    role { :tutor }
+    association :lecture
+
+    trait :tutor do
+      role { :tutor }
+    end
+
+    trait :editor do
+      role { :editor }
+    end
+
+    trait :teacher do
+      role { :teacher }
+    end
+
```

Comment:
We don't have a trait `:speaker` here. Lines like `build(:voucher, :speaker, lecture: seminar)` only work because `role` is the first parameter.

Reply:
I added the missing trait.